OZON.ru.js

```diff
@@ -1,396 +1,120 @@
-			break;
-		case "Переводчик":
-		case "Переводчики":
-			creatorsObject["translators"] = GetCreators(value, "translator");
-			break;
-		case "Редактор":
-		case "Редакторы":
-			creatorsObject["editors"] = GetCreators(value, "editor");
-			break;
-		case "Язык":
-		case "Языки":
-			result["language"] = GetLanguage(value);
-			break;
-		case "Издательство":
-			result["publisher"] = value;
-			break;
-		case "Серия":
-			result["series"] = value;
-			break;
-		case "ISBN":
-			result["ISBN"] = GetISBN(value);
-			result["date"] = GetDate(value);
-			break;
-		default:
-			//"Naked" year
-			if (name.match(/^\d\d\d\d$/) && !result["date"]) {
-				result["date"] = name;
-			}
+		if (/(Редакторы?|Составител[ьи])/.test(property.name)) {
+			getMultipleCreators(property.value, 'editor').forEach(function (d) { book.creators.push(d); });
 		}
+		if (property.name === 'Серия') book.series = property.value;
 	}
-	creatorsObject["all"] = new Array();
-	if(creatorsObject["authors"]) {
-		creatorsObject["all"] =
-		creatorsObject["all"].concat(creatorsObject["authors"]);
-	}
-	if(creatorsObject["compilers"]) {
-		creatorsObject["all"] =
-		creatorsObject["all"].concat(creatorsObject["compilers"]);
-	}
-	if(creatorsObject["translators"]) {
-		creatorsObject["all"] =
-		creatorsObject["all"].concat(creatorsObject["translators"]);
-	}
-	if(creatorsObject["editors"]) {
-		creatorsObject["all"] =
-		creatorsObject["all"].concat(creatorsObject["editors"]);
-	}
-	if(creatorsObject["all"]) {
-		result["creators"] = creatorsObject["all"];
-	}
-	return result;
-}
 
-function ParseTechDesc(nameXPathRes, valueXPathRes) {
-	var result = Object();
-	var nameDOMNode;
-	var valueDOMNode;
-	while( (nameDOMNode = nameXPathRes.iterateNext()) &&
-	(valueDOMNode = valueXPathRes.iterateNext()) ) {
-		var nameText = CleanText(nameDOMNode.textContent);
-		var valueText = CleanText(valueDOMNode.textContent);
-		if (nameText == "Страниц") {
-			result["numPages"] = GetNumPages(valueText);
-			return result;
-		}
-	}
+	book.complete();
 }
 
-function ParseDetailDesc(nameXPathRes, valueXPathRes) {
-	var result = Object();
-	var nameDOMNode;
-	var valueDOMNode;
-	while( (nameDOMNode = nameXPathRes.iterateNext()) &&
-	(valueDOMNode = valueXPathRes.iterateNext()) ) {
-		var nameText = CleanText(nameDOMNode.textContent);
-		var valueText;
-		switch(nameText) {
-		case "От производителя":
-			valueText =
-			valueDOMNode.innerHTML.replace(/^[\s\S]*ANNOTATION\] -->/, "");
-			//Zotero.debug("valueText -> " + valueText);
-			valueText = Zotero.Utilities.cleanTags(valueText);
-			result["abstractNote"] = valueText;
-			break;
-		case "От OZON.ru":
-			valueText =
-			valueDOMNode.textContent.replace(/^[\s\S]*От OZON.ru/, "");
-			valueText = CleanText(valueText);
-			result["edition"] = GetEdition(valueText);
-			break;
-		}
+function getAdditionalProperty(element) {
+	// cyrillic alphabet
+	var matches = /[\u0430-\u044f\u0410-\u042fёЁ]+(?=:)/.exec(element.textContent);
```

Comment:
nit: order ranges in order of code point values, though in this case the two ranges are adjacent, so they can be just a single range.